plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -242,6 +242,133 @@ function MenuDialog:init()
+                fields = {
+                    {
+                        description = "Review intervals",
+                        text = current_intervals,
+                        hint = _("10m 12h 1d"),
+                    },
+                    {
+                        description = "Interval modifier",
+
+                        text = tostring(interval_modifier),
+                        hint = _("2")
+                    },
+                },
+                buttons = {
+                    {
+                        {
+                            text = _("Close"),
+                            id = "close",
+                            callback = function()
+                                UIManager:close(self.settings_dialog)
+                            end,
+                        },
+                        {
+                            text = _("Help"),
+                            id = "help",
+                            callback = function()
+                                local helpMessage = InfoMessage:new{
+                                    text = [[Review intervals are the initial periods of time between reviews of words.
+
+Interval modifier is used after all review intervals have been completed to increase the time between reviews.
+
+For example, a word could be reviewed at 5m, 30m and 1d intervals before the interval modifier takes over with 2d, 4d, etc.]],
+
+                                    show_icon = true,
+                                    icon = "notice-info",
+                                }
+                                UIManager:show(helpMessage)
+                            end
+                        },
+                        {
+                            text = _("Default"),
+                            id = "default",
+                            callback = function()
+                                settings.review_intervals = nil
+                                settings.interval_modifier = nil
+                                saveSettings()
+                                UIManager:close(self.settings_dialog)
+                            end,
+                        },
+                        {
+                            text = _("Save"),
+                            is_enter_default = true,
+                            callback = function()
+                                local fields = self.settings_dialog:getFields()
+                                local interval_input = fields[1]
+                                local modifier_input = fields[2]
+
+                                local isIntervalValid = MenuDialog.isIntervalStringValid(interval_input)
+                                local isModifierValid = MenuDialog.isValidModifier(modifier_input)
+
+                                local intervalsArray = {};
+                                if isIntervalValid then
+                                    for inputTime in string.gmatch(interval_input, "%d+[m|d|h]") do
```

Comment:
You are duplicating the regex matching once in `isIntervalStringValid` and once here. Maybe do it just once?